blocksuite/affine/shared/src/viewport-renderer/renderer-utils.ts

```diff
@@ -23,58 +29,109 @@ export function syncCanvasSize(canvas: HTMLCanvasElement, host: HTMLElement) {
   canvas.style.pointerEvents = 'none';
 }
 
+function clientToModelCoord(
+  viewport: ViewportRecord,
+  clientCoord: [number, number]
+) {
+  const { left, top, viewportX, viewportY, zoom, viewScale } = viewport;
+
+  const [clientX, clientY] = clientCoord;
+  const viewportInternalX = clientX - left;
+  const viewportInternalY = clientY - top;
+  const modelX = viewportX + viewportInternalX / zoom / viewScale;
+  const modelY = viewportY + viewportInternalY / zoom / viewScale;
+  return [modelX, modelY];
+}
+
+function getSurfaceBlock(doc: Store) {
```

Comment:
We have at least 4 same function. Can you search before adding new one?

Reply:
We could just use `gfx.surface` if the `std` is accessible.

Reply:
Thanks for the advice. Current `getSurfaceBlock` impls have been unified and lifted up to `block-std` to be reused. I think we can remove other similar impls by using `gfx.surface` (they are actually querying `SurfaceComponent` rather than `SurfaceModel`), this refactoring should be responsible for standalone PRs.